x/gamm/pool-models/stableswap/pool.go

```diff
@@ -69,6 +78,45 @@ func (pa Pool) getPoolAmts(denoms ...string) ([]sdk.Int, error) {
 	return result, nil
 }
 
+// getScaledPoolAmts returns scaled amount of pool liquidity based on each asset's precisions
+func (pa Pool) getScaledPoolAmts(denoms ...string) ([]sdk.Int, error) {
+	result := make([]sdk.Int, len(denoms))
+	poolLiquidity := pa.PoolLiquidity
+	liquidityIndexes := pa.getLiquidityIndexMap(poolLiquidity)
+
+	for i, denom := range denoms {
+		liquidityIndex := liquidityIndexes[denom]
+
+		amt := poolLiquidity.AmountOf(denom)
+		if amt.IsZero() {
+			return []sdk.Int{}, fmt.Errorf("denom %s does not exist in pool", denom)
+		}
+		scalingFactor := pa.GetScalingFactorByLiquidityIndex(liquidityIndex)
+		result[i] = amt.QuoRaw(int64(scalingFactor))
+	}
+	return result, nil
+}
+
+// getDescaledPoolAmts gets descaled amount of given denom and amount
+func (pa Pool) getDescaledPoolAmt(denom string, amount sdk.Dec) sdk.Dec {
+	poolLiquidity := pa.PoolLiquidity
+
+	liquidityIndexes := pa.getLiquidityIndexMap(poolLiquidity)
+	liquidityIndex := liquidityIndexes[denom]
+
+	scalingFactor := pa.GetScalingFactorByLiquidityIndex(liquidityIndex)
+	return amount.MulInt64(int64(scalingFactor))
+}
+
+// getLiquidityIndexMap creates a map of denoms to its index in pool liquidity
+func (pa Pool) getLiquidityIndexMap(poolLiquidity sdk.Coins) map[string]int {
```

Comment:
suggest removing `poolLiquidity` from parameters, if possible, since it can be grabbed from the receiver